src/components/ColoredTd.tsx

```diff
@@ -1,6 +1,6 @@
 import { Td, TdProps } from '@patternfly/react-table';
 import { createUseStyles } from 'react-jss';
-import { choosableColors, Customer, getCustomers } from 'src/api/CustomerApi';
+import { Color as ApiColor } from 'src/api/CustomerApi';
```

Comment:
So the source of your error is you were attempting to import something with the same name as a variable you created below "Color". 

You need to delete the Color type you created below (as it won't be used anyway), and import Color from above. 
Not Color "as" somethingElse